src/mgr/InfoAction.h

```diff
@@ -25,8 +25,8 @@ class InfoActionData
     InfoActionData& operator += (const InfoActionData& stats);
 
 public:
-    struct timeval squid_start = {};
-    struct timeval current_time = {};
+    struct timeval squid_start = {0, 0};
+    struct timeval current_time = {0, 0};
```

Comment:
This is in reaction to a version of gcc failing the build (IIRC it is Centos-7's). I was as surprised as you.

Reply:
Please note that your statement does not answer my two questions. I think it is important to get to the bottom of this because consistent initialization (across the Squid code base) may depend on finding out the truth. The current PR uses inconsistent initialization approaches. If we can keep initialization consistent, we should. If we cannot, we should know why we can't.

Reply:
gcc on centos-7 is failing in ext_session_acl.cc, in fact

```
/home/kinkie/src/squid/src/acl/external/session/ext_session_acl.cc: In function 'void session_login(char*, size_t)':
/home/kinkie/src/squid/src/acl/external/session/ext_session_acl.cc:226:21: error: missing initializer for member 'TDB_DATA::dptr' [-Werror=missing-field-initializers]
     DB_ENTRY key = {};
                     ^
/home/kinkie/src/squid/src/acl/external/session/ext_session_acl.cc:226:21: error: missing initializer for member 'TDB_DATA::dsize' [-Werror=missing-field-initializers]
/home/kinkie/src/squid/src/acl/external/session/ext_session_acl.cc:227:22: error: missing initializer for member 'TDB_DATA::dptr' [-Werror=missing-field-initializers]
     DB_ENTRY data = {};
                      ^
/home/kinkie/src/squid/src/acl/external/session/ext_session_acl.cc:227:22: error: missing initializer for member 'TDB_DATA::dsize' [-Werror=missing-field-initializers]
/home/kinkie/src/squid/src/acl/external/session/ext_session_acl.cc: In function 'void session_logout(char*, size_t)':
/home/kinkie/src/squid/src/acl/external/session/ext_session_acl.cc:247:21: error: missing initializer for member 'TDB_DATA::dptr' [-Werror=missing-field-initializers]
     DB_ENTRY key = {};
                     ^
/home/kinkie/src/squid/src/acl/external/session/ext_session_acl.cc:247:21: error: missing initializer for member 'TDB_DATA::dsize' [-Werror=missing-field-initializers]
At global scope:
```

centos-8 has no problems with that code

Reply:
According to quick [tests](https://cpp.godbolt.org/z/HBfwBe), GCC v5 is required to process empty initialization lists without warnings when given `-Wmissing-field-initializers`. GCC v4 is not enough. In those tests, GCC v4 initializes the values correctly despite the warnings. We still want to support GCC v4 builds. Official Squid already uses `{}` initializers. Using `{}` is better.

Given all these facts, I suggest:

1. Disable `-Wmissing-field-initializers` for GCC v4.
2. Use `{}` initializers in new and changed code.

Reply:
It turns out that -Wextra does bad things to gcc-4, it starts throwing the most bizarre errors in the constructor of HttpRequestMethod , so I've diasbled it all there.

Reply:
Sounds good to me! I think disabling any _harmful_ warnings, including `-Wextra` for GCC v4 (or any old compiler) is the right thing to do. We want to lint sources with the latest (presumably "best") compiler in each compiler series. Everything else is optional and, if it is in the way, should be disabled. Working very hard on supporting a GCC v4-specific warning is essentially a waste of resources.

Reply:
To avoid misunderstanding, please note that disabling `-Wextra` for GCC v4 is perfectly compatible with my [suggestion](https://github.com/squid-cache/squid/pull/667#discussion_r445574389) on this thread, and the second item in that suggestion remains valid (and probably requires some PR changes):

> 2. Use `{}` initializers in new and changed code

Reply:
> 1. Use `{}` initializers in new and changed code

Right. Changing some more; I'll recheck if I got them all